Src/Plugins/Security/DomainFilteringValidator.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security.HelpersUtiliesAndExtensions;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public abstract class DomainFilteringValidator : ValidatorBase
+    {
+        public static string IsValidStatic(DomainFilteringValidator validator,
+                                           ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            validator.MatchCleanup(ref matchedPattern,
```

Comment:
i see what you've done. we can clean this up. Perhaps you could simply put the validator host exclusion helper in the base class. Or you could add the match clean-up helper in this class, rather than pushing it into the base class.

Honestly, if all of this is just to get the HostExclusion helper into the base class, suggest just doing that. and incorporate this utility into the derived types as required. this is in line with my philosophy of implementing as little object hierarchy, etc., as is required.

Reply:
Fixed as discussed.

---
In reply to: [575506264](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575506264) [](ancestors = 575506264)